### DISCUSSION TITLE
using a keg-only dependency for a brew install cmake process

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#2498] [Tap maintenance and Homebrew development]
i've been working on getting `brew install freecad` to an actual workable state, however one of the latest issues i'm running into is the cmake process that the `brew install --verbose freecad` kicks off does not manage to find the path / binaries for `pyside@2`.

freecad has several workbenches and runtime deps that require `pyside@2` to be picked up durning the configure step of the installation. the _cmake, make, make install_ processes will continue without knowning the path to the pyside@2 binaries required for runtime. however, after the install process completes on initial launch of freecad the `$USER` will be greeted with something like the below,

---

```
14:55:05  During initialization the error "No module named 'PySide2'" occurred in /Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/Draft/InitGui.py
14:55:05  Please look into the log file for further information
14:55:07  During initialization the error "No module named 'PySide2'" occurred in /Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/Arch/InitGui.py
14:55:07  Please look into the log file for further information
14:55:07  During initialization the error "No module named 'PySide2'" occurred in /Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/AddonManager/InitGui.py
14:55:07  Please look into the log file for further information
14:55:07  During initialization the error "No module named 'PySide2'" occurred in /Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/Tux/InitGui.py
14:55:07  Please look into the log file for further information
14:55:09  Unknown command 'Std_AddonMgr'
14:55:09  Traceback (most recent call last):
  File "<string>", line 2, in <module>
  File "/Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/Start/StartPage/StartPage.py", line 29, in <module>
    from . import TranslationTexts
  File "/Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/Mod/Start/StartPage/TranslationTexts.py", line 24, in <module>
    from PySide import QtGui
  File "/Applications/freecads/brewmaster.brew.install/FreeCAD.app/Contents/MacOS/PySide/__init__.py", line 2, in <module>
    from PySide2 import __version__
<class 'ModuleNotFoundError'>: No module named 'PySide2'
14:55:09  No module named 'PySide2'
```

![image](https://user-images.githubusercontent.com/613805/142690853-0cb2a2e6-bba2-40f8-95cb-c66615832889.png)

---

in contrast, if i run `brew link pyside@2` **before** running `brew install freecad` then the _cmake_ process will pickup/detect pyside@2 binaries required for bundling into the `FreeCAD.app` bundle during the `make install` process of the `brew install freecad` command/process.

...that's good to know but doesn't seem to help when requiring pyside@2 to be linked using github actions.

please correct if i'm wrong, but since `pyside@2` is `keg-only` brew _test-bot_ will not link pysdie@2 during the steps of building freecad.

i attemtped adding the `.bin` prefix path with the following line in the `freecad.rb` formula, and the reran `brew install freecad` but cmake still did not detect the `pyside@2` installation / path to required runtime binaries to bundle into the `FreeCAD.app` bundle.

```
ENV.prepend_path "PATH", Formula["pyside@2"].bin
```

i've read the formula cookbook several times, but haven't come across a use case in there that could help me resolve this issue that is arising.

as always, any and all help would be greatly appreciated 🙏

**Comment by carlocab** [comment]
It's a bit weird the CMake doesn't find `pyside@2` during the build, since usually CMake is pretty good at finding keg-only dependencies.

Try doing
```ruby
ENV.prepend "CMAKE_PREFIX_PATH", Formula["pyside@2"].opt_lib/"cmake"
```
in the formula. If that doesn't help (it might not, since this path should already be in `CMAKE_PREFIX_PATH`), then it might help to manually set `PYSIDE2UICBINARY` and `PYSIDE2RCCBINARY` by passing them with `-DPYSIDE2UICBINARY=foo` in your `cmake` invocation.

Finally, failing that, it seems like it's just a matter of finding Python modules, so it might work to just find a way to expose the environment variable `PYTHONPATH="$(brew --prefix pyside@2)/lib/python3.9/site-packages"` to your freecad binary at runtime.
